### PR TITLE
feat: test workflows

### PR DESCRIPTION
**Pull request by junminahn**

### CODE REVIEW

File: transition-scripts/helpers/keycloak.sh [hunk 1]
```diff
@@ -32,3 +32,191 @@ wait_for_keycloak_all_ready() {
...
+    "$(get_ocp_keycloak_url "$cluster" "$namespace")/auth/admin/realms/$realmname")
+
+  if [ "$status_code" -ne "204" ]; then
+    return
+  fi
+
+  echo "$data"
+}
+
+remove_all_realms() {
```
Comment (thegentlemanphysicist) [inline transition-scripts/helpers/keycloak.sh:205]: Does this mean each time the tests are run the sandbox dev deployment will be nuked of realms?

File: transition-scripts/values/values-gold-eb75ad-dev-standby.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+replicaCount: 0
```
[the remark this answers is not in the document]
Comment (junminahn) [reply transition-scripts/values/values-gold-eb75ad-dev-standby.yaml:1]: no, in standby, there will be no Keycloak pods running.

File: transition-scripts/helpers/github.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+GITHUB_ORG="bcgov"
+GITHUB_REPO="sso-switchover-agent"
+
+trigger_github_dispatcher() {
```
[the remark this answers is not in the document]
Comment (junminahn) [reply transition-scripts/helpers/github.sh:6]: these helper functions are only used for the test script for now.

File: transition-scripts/helpers/keycloak.sh (hunk 1 above)
Comment (junminahn) [reply transition-scripts/helpers/keycloak.sh:205]: yes, as each test would create 10 realms, it needs to remove all test realms before it runs.

File: transition-scripts/helpers/keycloak.sh [hunk 4]
```diff
@@ -32,3 +32,191 @@ wait_for_keycloak_all_ready() {
...
+  namespace="$2"
+
+  count=0
+  wait_keycloak_up() {
+    status=$(check_keycloak_health "$cluster" "$namespace")
+    info "checking keycloak app health - $status"
+
+    if [ "$status" == "up" ]; then return 1; fi
+
+    if [[ "$count" -gt 50 ]]; then
```
[the remark this answers is not in the document]
Comment (junminahn) [reply transition-scripts/helpers/keycloak.sh]: I updated to wait for 20 mins.

[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply transition-scripts/helpers/keycloak.sh]: thanks

**Review by thegentlemanphysicist — APPROVED** [review]

**Pull request merged**